platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/CryptographyHolder.java

```diff
@@ -48,30 +54,44 @@ private CryptographyHolder() {
     }
 
     /**
-     * By calling the method the interal {@link Cryptography} instance will be removed and a new instance will be
-     * created when {@link #get()} is called the next time.
+     * Setup the config holder.
+     *
+     * @param cryptography the {@link CryptoEngine} to use
      */
-    public static void reset() {
+    public static void set(@NonNull final Cryptography cryptography) {
         try (final Locked ignored = lock.lock()) {
-            cryptography = null;
+            CryptographyHolder.cryptography = cryptography;
         }
     }
 
+    /**
+     * Setup cryptography. Intended only to be invoked by unit tests.
+     */
+    private static void defaultTestSetup() {
+        // This exception is intended to intentionally fail validators for deployments that do not set up
+        // cryptography correctly. This won't cause a problem during unit tests, which are the biggest offender
+        // w.r.t. not setting up cryptography correctly.
+        logger.error(EXCEPTION.getMarker(), "CryptographyHolder not initialized, using default config");
+
+        final Configuration defaultConfiguration = ConfigurationBuilder.create()
+                .withConfigDataType(CryptoConfig.class)
+                .build();
+
+        cryptography = Cryptography.create(defaultConfiguration);
+    }
+
     /**
      * Getter for the {@link Cryptography} singleton.
      *
      * @return the {@link Cryptography} singleton
      */
+    @NonNull
     public static Cryptography get() {
-        if (cryptography == null) {
-            try (final Locked ignored = lock.lock()) {
-                if (cryptography == null) {
-                    cryptography = new CryptoEngine(
-                            getStaticThreadManager(),
-                            ConfigurationHolder.getInstance().get().getConfigData(CryptoConfig.class));
-                }
+        try (final Locked ignored = lock.lock()) {
```

Comment:
you do not need to lock if cryptography != null

Reply:
It's necessary if we want to be thread safe. If thread A sets the variable and thread B reads it without establishing a happens-before relationship with thread A, then the variable it reads may be partially initialized.